File: backend/internal/models/project.go
package models

import (
	"time"
)

type Project struct {
	ID              uint      `json:"id" gorm:"primaryKey"`
	Title           string    `json:"title"`
	Location        string    `json:"location"`
	LocationMapLink string    `json:"location_map_link"`
	Owner           string    `json:"owner"`
	Category        string    `json:"category"`
	Images          []string  `json:"images" gorm:"serializer:json"` // JSON serializer for array storage
	Description     string    `json:"description"`
	Status          string    `json:"status"`
	SortOrder       int       `json:"sort_order" gorm:"default:0"`
	IsActive        bool      `json:"is_active" gorm:"default:true"`
	CreatedBy       string    `json:"created_by"`
	UpdatedBy       string    `json:"updated_by"`
	CreatedAt       time.Time `json:"created_at"`
	UpdatedAt       time.Time `json:"updated_at"`
}

File: frontend/src/types/project.ts
export interface Project {
    id: number;
    title: string;
    location: string;
    location_map_link: string;
    owner: string;
    category: string;
    images: string[];
    description: string;
    status: string;
    sort_order: number;
    is_active: boolean;
    created_by: string;
    updated_by: string;
    created_at: string;
    updated_at: string;
}

export interface Category {
    id: number;
    name: string;
    slug: string;
    sort_order: number;
    is_active: boolean;
    created_at: string;
    updated_at: string;
}

export interface CreateProjectInput {
    title: string;
    location: string;
    location_map_link?: string;
    owner?: string;
    category: string;
    images: string[];
    description?: string;
    status?: string;
    sort_order?: number;
    is_active?: boolean;
}

export interface UpdateProjectInput extends Partial<CreateProjectInput> {}

export interface CreateCategoryInput {
    name: string;
    slug?: string;
    sort_order?: number;
    is_active?: boolean;
}

export interface UpdateCategoryInput extends Partial<CreateCategoryInput> {}

export interface UploadResult {
    key: string;
    url: string;
    thumbnail_key: string;
    thumbnail_url: string;
}
